### PR TITLE
Leaderboard

### PR DESCRIPTION
**Pull request by K1llMan**
image

### CODE REVIEW

File: content/panorama/scripts/custom_game/leaderboard.js [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+}
+
+(function () {
+    let players = [];
+    for (let i = 0; i < 100; i++)
+        players.push({
+            rank: 20,
+            steamId: 76561197988355984,
+            rating: 1000
+        });
```
Comment (ark120202) [inline content/panorama/scripts/custom_game/leaderboard.js:45]: Before merge it should be changed to the real data from https://github.com/dota2unofficial/12v12/blob/master/game/scripts/vscripts/common/webapi.lua#L108

```lua
CustomNetTables:SetTableValue("game_state", "leaderboard", data.leaderboard)
```

File: content/panorama/scripts/custom_game/common.js [hunk 2]
```diff
@@ -0,0 +1,45 @@
+let getHUDRoot = () => {
```
[the remark this answers is not in the document]
Comment (K1llMan) [reply content/panorama/scripts/custom_game/common.js:1]: Global access and we need UI framework for common functions and use cases.

File: content/panorama/styles/custom_game/common.css [hunk 3]
```diff
@@ -0,0 +1,27 @@
+.BackgroundPanel {
```
[the remark this answers is not in the document]
Comment (K1llMan) [reply content/panorama/styles/custom_game/common.css:1]: idk about it. Is there any style guidelines for consistent UI?

**Comment by K1llMan** [comment]
Fixed.

Strings added into addon_english
		// Leaderboard
		"leaderboard"						"Leaderboard"
		"leaderboard_rank"					"Rank"
		"leaderboard_player"				"Player"
		"leaderboard_rating"				"Rating"

**Comment by K1llMan** [comment]
Updated.

File: game/scripts/vscripts/common/webapi.lua [hunk 4]
```diff
@@ -103,9 +103,12 @@ function WebApi:BeforeMatch()
 				wins = player.wins,
 				loses = player.loses,
 			}
+
+
```
Comment (ark120202) [inline game/scripts/vscripts/common/webapi.lua:107]:
```suggestion
```

**Pull request merged**